### PR TITLE
Grant availability metrics script

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: tests/metrics/grant_availability.py
Comment (vzotova) [inline tests/metrics/grant_availability.py:54]: maybe use `mktmpdir` and `Path`?

File: tests/metrics/grant_availability.py
Comment (arjunhassard) [inline tests/metrics/grant_availability.py:85]:
```suggestion
    """Perform a granting operation for metrics collection."""
```

File: tests/metrics/grant_availability.py
Comment (arjunhassard) [inline tests/metrics/grant_availability.py:159]:
```suggestion
    """Prepares the filesystem and logger for grant metrics collection"""
```

File: tests/metrics/grant_availability.py
Comment (arjunhassard) [inline tests/metrics/grant_availability.py:174]: This helps developers/Alices discover the availability of a subset of Ursulas that they specify in a URI list?

**Review by arjunhassard — APPROVED** [review]
👌

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py:52]: Haha - yes - let's move this to an envvar. :sweat_smile:

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py:174]: @arjunhassard - Yes, exactly.

**Review by cygnusv — APPROVED** [review]
Let's move this forward :)

File: tests/metrics/grant_availability.py
Comment (derekpierre) [inline tests/metrics/grant_availability.py:122]: I think we are running once more than expected and the check should be `(i+1) == iterations`. We increment `i` after we run i.e. i = 0 is really iteration 1

**Review by derekpierre — CHANGES_REQUESTED** [review]

**Comment by KPrasch** [comment]
Error captured in a resent result (blocked by #1861 and friends) : https://dpaste.com/AZ8DYD36G

File: tests/metrics/grant_availability.py
Comment (derekpierre) [reply tests/metrics/grant_availability.py:52]: Still planning on doing an envvar here?

File: tests/metrics/grant_availability.py
Comment (derekpierre) [reply tests/metrics/grant_availability.py:54]: I like @vzotova's suggestion here.

File: tests/metrics/grant_availability.py
Comment (derekpierre) [reply tests/metrics/grant_availability.py:122]: Are we running on extra time here?

Lets say you want 1 iteration.

Loop 1: i = 0, perform metric grant, 0 != 1 so continue the loop, increment i to 1
Loop 2: i = 1, perform metric grant, 1 == 1 so stop.

**Review by derekpierre — COMMENTED** [review]
@KPrasch comments to clarify my concerns.

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py:122]: I knew what you meat - thanks for explanation though :-)

I think I forgot to add these two lines to my last patch.

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py:52]: Heh - my computer froze when pushing, standby.

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py:119]: I don't see why not... the more information we can extract the better IMO.

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py]: :heavy_check_mark:

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py]: :heavy_check_mark:

File: tests/metrics/grant_availability.py
Comment (KPrasch) [reply tests/metrics/grant_availability.py]: :heavy_check_mark:

File: tests/metrics/grant_availability.py
Comment (derekpierre) [reply tests/metrics/grant_availability.py:119]: Did you want to add the label to the text here?

**Review by derekpierre — APPROVED** [review]
🎸  - couple remaining minor comments (tmp directory creation, adding the label to the PEK output) but nothing to hold back the merge. Merge at will.

**Comment by KPrasch** [comment]
:+1: - I'm in no rush to merge this one - I'll keep adding to it until someone else merges it.

**Pull request merged**
